### PR TITLE
event info for registration

### PR DESCRIPTION
**Pull request by gloveboxes**

### CODE REVIEW

File: database/aoai-proxy.sql [hunk 1]
```diff
@@ -262,6 +262,33 @@ $$;
 
 ALTER FUNCTION aoai.get_attendee_authorized(p_event_code character varying, p_api_key uuid) OWNER TO admin;
```
Comment (aaronpowell) [inline database/aoai-proxy.sql:264]: Is there really any value to this being a function rather than just inline SQL?

**Comment by gloveboxes** [comment]
Yeah – protection against sql injection attacks. With Functions there is built in type checking and I’m using functions so want to be consistent

From: Aaron Powell ***@***.***>
Date: Friday, 12 January 2024 at 1:41 pm
To: microsoft/azure-openai-service-proxy ***@***.***>
Cc: Dave Glover (HE/HIM) ***@***.***>, Author ***@***.***>
Subject: Re: [microsoft/azure-openai-service-proxy] event info for registration (PR #117)

@aaronpowell commented on this pull request.

________________________________

In database/aoai-proxy.sql<https://github.com/microsoft/azure-openai-service-proxy/pull/117#discussion_r1449726947>:

> @@ -262,6 +262,33 @@ $$;



 ALTER FUNCTION aoai.get_attendee_authorized(p_event_code character varying, p_api_key uuid) OWNER TO admin;



Is there really any value to this being a function rather than just inline SQL?

**Comment by gloveboxes** [comment]
I'm using functions across the board - built in type checking and protection again sql injection attacks.

**Review by aaronpowell — COMMENTED** [review]
SQL injection protection would be handled by the Python pg library, as it's how the parameters are added to the statement that is built up.

What do you mean by type checking? The function is evaluated at runtime, meaning that if there's an incorrectly named column that won't be found until the function is executed, which happened with `add_event`: https://github.com/microsoft/azure-openai-service-proxy/commit/938b9f6a9abda8fd2ffde20b605d8e4c8b882b61#diff-cbb34fa98bc728478ae725b61790b34f4470e3587fb7b7682efa15bce46ca4faR81 is an example.

**Review by aaronpowell — APPROVED** [review]

**Pull request merged**